internal/policy/policy.go

```diff
@@ -715,6 +713,65 @@ func ListRules(ctx context.Context, repo *git.Repository) ([]*DelegationWithDept
 	return allDelegations, nil
 }
 
+func Apply(ctx context.Context, repo *git.Repository, signRSLEntry bool) error {
+	state, err := LoadCurrentState(ctx, repo, PolicyStagingRef)
+	if err != nil {
+		return fmt.Errorf("failed to load current state: %w", err)
+	}
+
+	if err := state.Verify(ctx); err != nil {
+		return fmt.Errorf("state verification failed: %w", err)
+	}
+
+	// Get the reference for the PolicyRef
+	policyRef, err := repo.Reference(plumbing.ReferenceName(PolicyRef), true)
+	if err != nil {
+		return fmt.Errorf("failed to get policy reference %s: %w", PolicyRef, err)
+	}
+
+	// Get the reference for then PolicyStagingRef
+	policyStagingRef, err := repo.Reference(plumbing.ReferenceName(PolicyStagingRef), true)
+	if err != nil {
+		return fmt.Errorf("failed to get policy staging reference %s: %w", PolicyStagingRef, err)
+	}
+
+	// Check if the PolicyStagingRef is ahead of PolicyRef (fast-forward)
+
+	policyStagingCommit, err := gitinterface.GetCommit(repo, policyStagingRef.Hash())
+	if err != nil {
+		// if there is no tip for the policy staging ref, this means that no change will be made to the policy ref
+		return fmt.Errorf("failed to get policy staging tip commit: %w", err)
+	}
+	policyCommit, err := gitinterface.GetCommit(repo, policyRef.Hash())
+	if err == nil {
+		// This check ensures that the policy staging branch is a direct forward progression of the policy branch,
+		// preventing any overwrites of policy history and maintaining a linear policy evolution, since a
+		// fast forward merge does not work with a non-linear history.
+
+		// This is only being checked if there are no problems finding the tip of the policy ref, since if there
+		// is no tip, then it cannot be an ancestor of the tip of the policy staging ref
+		isAncestor, err := policyCommit.IsAncestor(policyStagingCommit)
```

Comment:
gitinterface.KnowsCommit does this for us IIRC

Reply:
Fixed